filterapi/filterconfig.go

```diff
@@ -164,6 +164,34 @@ type Backend struct {
 	// Auth is the authn/z configuration for the backend. Optional.
 	// TODO: refactor after https://github.com/envoyproxy/ai-gateway/pull/43.
 	Auth *BackendAuth `json:"auth,omitempty"`
+	// DynamicLoadBalancing is the dynamic backend configuration which forces the AI filter to
+	// route the request to the specified endpoints (ip:port) instead of the cluster.
+	DynamicLoadBalancing *DynamicLoadBalancing `json:"dynamicBackend,omitempty"`
```

Comment:
I think this is a level above backend, think about the case you want to failover between different backends GCP, AWS for Anthropic model

Reply:
right!

Reply:
ok so the reality is such failover like "going through filter chain again when the request got 5xx" cannot be implemented with the inference extension's spec. But I agree this should be one level above to allow us to our own failover mechanism based on that. I will put a comment in #34